### ISSUE TITLE
snakemake loop issue

### ISSUE THREAD

**Issue by anyexingluzk** [PyPSA/pypsa-eur#31]
What I want to achieve is to run a loop in snakemake, giving the initial input file and generating the output iteratively. I took the approach using "for" loop, i.e.,
_rule all:
     input: "{wildcards}-3"
for i in range(n):
     rule:
         input: "{wildcards}-{param}".format(param=i-1)
         output: "{wildcards}-{param}".format(param=i)
         script: do something_
If the initial file is named after "A-0", the workflow is supposed to generate outputs until "A-3". However, I got KeyError: "wildcards", and I suspect it is due to the fact that wildcards and python string format cannot be used in the same line. Does anyone have solutions to this problem? Or maybe you have a better way to implement the loop?

**Comment by coroa** [comment]
Your issue looks very specific to snakemake probably their issue tracker is more suited to your question.

The error you are seeing is that format complains it does not know how to to fill in ‘wildcards’, but I think you know that. You could use {{wildcards}} instead.

But, I don’t think it’s possible to create rules in a loop.

It probably makes sense to describe what you are actually trying to achieve.

**Comment by anyexingluzk** [comment]
Actually, it worked without the wildcards as I have tested the following workflow,
_rule all:
input: "A-3"
for i in range(n):
rule:
input: "A-{param}".format(param=i-1)
output: "A-{param}".format(param=i)
script: do something_
when A-0 is given, it is able to generate A-1, A-2 until A-3. What I want to do is very much like this in principle. I did realize it is more relevant to snakemake issue, yet I wanted to ask if you have done similar before.
